spec/ruby/core/string/rpartition_spec.rb

```diff
@@ -11,6 +11,18 @@
     "hello".rpartition("hello").should == ["", "hello", ""]
   end
 
+  it "returns original string if regexp doesn't match" do
+    "hello".rpartition("/x/").should == ["", "", "hello"]
+  end
+
+  it "returns new object if doesn't match" do
+    "hello".rpartition("/no_match/").last.object_id.should_not eql("hello".object_id)
```

Comment:
`.last.should_not.equal?("hello")` is better here, I'll change to that.
Also we need to store the first `"hello"` in a variable because this file does not use frozen string literals.